api/app/Http/Controllers/ChildEntityController.php

```diff
@@ -290,82 +319,84 @@ public function deleteOne($id, $childId)
+            $childModels = $this->getManyChildEntitiesByIdsAndParent($ids,$model);
         }
 
+
         if (!empty($childModels) && $childModels->count() !== count($ids)) {
             throw $this->notFoundManyException($ids, $childModels, $childKey);
         }
 
-        /** @var BaseModel[] $childModels */
-        foreach ($childModels as $childModel) {
-            $childModel->markAsDeleted();
-        }
-
-        $model->setRelation($this->relationName, $childModels, $relation);
-        $this->saveModel($model);
+        ModelHelper::deleteMany($childModels);
 
         return $this->getResponse()->noContent();
     }
 
     /**
-     * @param $id
      * @return BaseModel
      */
-    protected function getParentModel($id)
+    public function getParentModel()
     {
-        $this->validateId($id, $this->getKeyName(), $this->validateRequestRule);
+        return $this->parentModel;
+    }
 
-        try {
-            return $this->getRepository()->find($id);
-        } catch (ModelNotFoundException $e) {
-            throw $this->notFoundException($this->getKeyName());
-        }
+    /**
+     * @return ChildBaseModel
+     */
+    public function getChildModel()
+    {
+        return $this->childModel;
     }
 
     /**
-     * @param BaseModel $model
-     * @throws \Exception
-     * @throws \Exception|null
-     * @throws \Spira\Repository\Repository\RepositoryException
+     * @param $id
+     * @return BaseModel
      */
-    protected function saveModel(BaseModel $model)
+    protected function getParentEntityById($id)
     {
-        try {
-            $this->getRepository()->save($model);
-        } catch (ValidationException $e) {
-            if ($exception = $model->getRelationErrors($this->relationName)) {
-                throw $exception;
-            }
+        return $this->getParentModel()->findOrFail($id);
+    }
 
-            throw $e;
-        }
+    /**
+     * @param $id
+     * @param $parent
+     * @return BaseModel
+     */
+    protected function getChildEntityByIdAndParent($id, $parent)
+    {
+        return $this->getChildModel()->findByIdAndParent($id,$parent);
     }
 
     /**
-     * @param BaseModel $model
-     * @return HasOneOrMany
+     * @param $ids
+     * @param $parent
+     * @return Collection
      */
-    protected function getRelation(BaseModel $model)
+    protected function getManyChildEntitiesByIdsAndParent($ids, $parent)
     {
-        return $model->{$this->relationName}();
+        return $this->getChildModel()->findManyByIdsAndParent($ids,$parent);
     }
 
     /**
-     * @param Relation $relation
-     * @return string
+     * @param $parent
+     * @return Collection
      */
-    protected function getChildKey(Relation $relation)
+    protected function getAllChildEntitiesByParent($parent)
     {
-        return $relation->getQuery()->getModel()->getKeyName();
+        return $this->getChildModel()->findAllByParent($parent);
     }
+
```

Comment:
I know, but what I am saying is that a model should have nothing to do with validation beyond defining it's rules. It is a concern of the controller. We should remove validation from the save method too. 

See http://laravel.com/docs/5.1/validation

Our controllers should be calling something like
```
$this->validate($request, $this->getModel()->getValidationRules());
```
right at the top of the controller. For the cases like patch where required might not make sense, then the rules array would be mutated to only include the keys that were in the request.